### PR TITLE
Preview available email.

### PR DESCRIPTION
**Pull request by waheedahmed**
[ECOM-6084](https://openedx.atlassian.net/browse/ECOM-6084)

As a Publisher tool owner, I'd like an email template created which will be used to send emails to Course Team when a Preview link has been added by the Publisher.

screen shot 2017-03-02 at 5 09 42 pm

### CODE REVIEW

**Comment by waheedahmed** [comment]
@tasawernawaz @awais786 Please review.

File: course_discovery/apps/publisher/api/serializers.py [hunk 1]
```diff
@@ -87,6 +87,7 @@ def validate(self, data):
 
         return validated_values
 
+    @transaction.atomic
```
Comment (bderusha) [inline course_discovery/apps/publisher/api/serializers.py:90]: Can you comment (here, not in the code) as to why this was added.  Just curious

File: course_discovery/templates/publisher/email/course_run/preview_available.html [hunk 2]
```diff
@@ -4,17 +4,17 @@
...
         {% blocktrans with link_start='<a href="' link_middle='">' link_end='</a>' trimmed %}
-            Preview for {{ link_start }}{{ page_url }}{{ link_middle }} {{ course_name }} {{ link_end }} is available for review.
+            A preview is now available for the {{ link_start }}{{ page_url }}{{ link_middle }} {{ course_run_number }} course run {{ link_end }} of {{ course_name }}. Visit {{ link_start }}{{ preview_link }}{{ link_middle }} link to preview {{ link_end }} to approve or decline the preview for this course run.
         {% endblocktrans %}
     </p>
 
     {% comment %}Translators: It's closing of mail.{% endcomment %}
     {% trans "Thanks," %}<br>
-    {% trans "The edX team" %}
+    edX {{ sender_role }}
```
Comment (bderusha) [inline course_discovery/templates/publisher/email/course_run/preview_available.html]: Same note here as the other PR. Please use dynamic site name here

**Review by bderusha — CHANGES_REQUESTED** [review]

File: course_discovery/apps/publisher/api/serializers.py (hunk 1 above)
Comment (waheedahmed) [reply course_discovery/apps/publisher/api/serializers.py:90]: @bderusha This was added if some error occurred during sending email, then user will not get an email notification and the user who is updating course_key or preview_url will think he has done his part but the course team user did not get any email, so he does not know a preview is available or studio instance is created for the course.

So I'm rolling back the whole transaction if some error occurred to let user know that there was an error during updating course_key or preview_url.

File: course_discovery/templates/publisher/email/course_run/preview_available.html (hunk 2 above)
Comment (waheedahmed) [reply course_discovery/templates/publisher/email/course_run/preview_available.html]: @bderusha Yeah it should be, but not mentioned in any ticket.

**Review by awais786 — APPROVED** [review]

File: course_discovery/templates/publisher/email/course_run/preview_available.html (hunk 2 above)
Comment (waheedahmed) [reply course_discovery/templates/publisher/email/course_run/preview_available.html]: I think we should create a ticket to fix it every where in publisher.

Comment (waheedahmed) [reply course_discovery/templates/publisher/email/course_run/preview_available.html]: Done.

**Comment by waheedahmed** [comment]
@bderusha Feedback addressed, please take another look.

**Review by bderusha — APPROVED** [review]

**Pull request merged**